### PR TITLE
feat: new `std/webgpu` sub-module

### LINKED ISSUE #3036: add `webgpu` folder for various WebGPU related utilities
This should then also move utilities from https://github.com/denoland/webgpu-examples

### PR DESCRIPTION
**Pull request by crowlKats**
Closes #3036

### CODE REVIEW

**Comment by iuioiua** [comment]
Is this still something you want to pursue, @crowlKats?

**Comment by iuioiua** [comment]
@crowlKats, do you think we should close this for now since the WebGPU API is currently not supported? We could re-open this and #3036 once support has been re-added.

**Comment by crowlKats** [comment]
@iuioiua i think webgpu coming back might be quite soon, so don't think it's necessary to close it

**Comment by iuioiua** [comment]
@crowlKats, it looks like `create_capture.ts` and `describe_texture_format.ts` might need to be removed to get the first pass over the line. Can you please confirm?

**Review by iuioiua — CHANGES_REQUESTED** [review]
I'm not familiar with the WebGPU API, but from a layout and structure perspective, mostly LGTM! I think we just missing a couple of pieces of documentation, mostly examples.

File: test.ts
Comment (iuioiua) [inline test.ts:1]: Is this here on purpose?

File: webgpu/mod.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * Utilities for interacting with the
+ * [WebGPU API]{@link https://developer.mozilla.org/en-US/docs/Web/API/WebGPU_API}.
+ */
+
+export * from "./create_capture.ts";
+export * from "./describe_texture_format.ts";
+export * from "./row_padding.ts";
+export * from "./texture_with_data.ts";
+export * from "./create_capture.ts";
+export * from "./describe_texture_format.ts";
```
Comment (iuioiua) [inline webgpu/mod.ts:13]:
```suggestion
```

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by iuioiua — APPROVED** [review]
LGTM! Niiiice. We should request review from someone who better knows the WebGPU API too.

**Comment by crowlKats** [comment]
@iuioiua there is no one in the core team like that 😅

**Comment by iuioiua** [comment]
Reviews from contributors are also welcome!

**Comment by iuioiua** [comment]
Note to maintainers: only merge this right before the next release. CI will remain in a failed state until the next runtime release.

**Comment by kt3k** [comment]
> Note to maintainers: only merge this right before the next release. CI will remain in a failed state until the next runtime release.

This is very tricky. [The current release flow](https://github.com/denoland/deno/blob/main/tools/release/release_doc_template.md) doesn't allow such thing because the next version of std needs to be released before the next version of CLI.

If we'd prefer to release this for the next version, maybe we should disable stable CI for a moment?

**Comment by load1n9** [comment]
LGTM nothing really needs to be changed atm

**Comment by crowlKats** [comment]
Thanks for reviewing @load1n9

**Comment by iuioiua** [comment]
PTAL @kt3k

**Review by kt3k — APPROVED** [review]
LGTM!

**Pull request merged**